### PR TITLE
Geomap: use same style config for makers and geojson

### LINKED ISSUE #41838: Geomap: Use standard StyleEditor to handle Geojson layer styles
Replace current Geojson style handling (for style rules) with modified version of universal StyleEditor component created in https://github.com/grafana/grafana/pull/41716

### PR DESCRIPTION
**Pull request by ryantxu**
This uses the same config object and UI for markers and geojson.

It works well -- but the UI is pretty gnarly :grimacing: 

**Which issue(s) this PR fixes**:

Fixes #41838 #41512

### CODE REVIEW

File: public/app/plugins/panel/geomap/layers/data/geojsonMapper.ts [hunk 1]
```diff
@@ -68,30 +77,39 @@ export const geojsonMapper: MapLayerRegistryItem<GeoJSONMapperConfig> = {
...
+          const s = await getStyleConfigState(r.style);
+          styles.push({
+            point: s.maker(s.base),
+            poly: polyStyle(s.base),
+            rule: r.rule,
+          });
+        }
+      }
+    }
+    if(true) {
```
Comment (nmarrs) [inline public/app/plugins/panel/geomap/layers/data/geojsonMapper.ts:93]: I think something may be up with our linter / maybe something is messed up locally for you with applying prettier? This didn't seem to break the PR build

File: public/app/plugins/panel/geomap/layers/data/geojsonMapper.ts [hunk 2]
```diff
@@ -129,10 +147,20 @@ export const geojsonMapper: MapLayerRegistryItem<GeoJSONMapperConfig> = {
...
             defaultValue: [],
-          });
+          })
+          .addCustomEditor({
+            id: 'config.style',
+            path: 'config.style',
+            name: 'Default Style',
+            description: 'The style to apply when no rules above match',
+            editor: StyleEditor,
+            settings: {},
```
Comment (nmarrs) [inline public/app/plugins/panel/geomap/layers/data/geojsonMapper.ts:161]: I'm guessing we will want to customize both of these StyleEditor instances (at this level and at style rule editor) in a similar way, correct? (i.e. remove field options in favor of fixed, etc.)

File: public/app/plugins/panel/geomap/layers/data/geojsonMapper.ts (hunk 1 above)
Comment (ryantxu) [reply public/app/plugins/panel/geomap/layers/data/geojsonMapper.ts]: hymmm -- that is scarry.  but I tihnk you are right!

File: public/app/plugins/panel/geomap/layers/data/geojsonMapper.ts (hunk 2 above)
Comment (ryantxu) [reply public/app/plugins/panel/geomap/layers/data/geojsonMapper.ts]: yes -- I just updated with a *super* simple version

File: public/app/plugins/panel/geomap/layers/data/StyleEditor.tsx [hunk 3]
```diff
@@ -59,7 +79,45 @@ export const StyleEditor: FC<StandardEditorProps<StyleConfig, any, any>> = ({ va
...
+                } as any
+              }
+            />
+          </InlineField>
+        </InlineFieldRow>
+      </>
+    );
+  }
+
+  const hasTextLabel = styleUsesText(value);
```
Comment (nmarrs) [inline public/app/plugins/panel/geomap/layers/data/StyleEditor.tsx:120]: We may also want to update `markersLayer` line 77 implementation of this property as well to use this new helper function as part of these changes

**Review by nmarrs — COMMENTED** [review]
UI looks a lot better! Only major thing was UX with entering feature property (losing focus after every key press)

**Review by nmarrs — APPROVED** [review]

**Pull request merged**